### PR TITLE
[TensorExpr] Delet `DimArg` class.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72390
* #72389

This class didn't add much value and only caused more boilerplate code.
This change removes the class and updates all the use cases with
uses of `ExprHandle`.

A side effect of this change is different names in loop variables, which
caused massive mechanical changes in our tests.

Differential Revision: [D34030296](https://our.internmc.facebook.com/intern/diff/D34030296)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).

File: torch/csrc/jit/tensorexpr/expr.cpp [hunk 1]
```diff
@@ -430,13 +430,13 @@ BufHandle Buf::make(const std::vector<ExprHandle>& dims, Dtype dtype) {
...
-      c10::optional<std::vector<ExprHandle>> strides,
-      c10::optional<ExprHandle> qscale,
-      c10::optional<ExprHandle> qzero) {
+    const std::string& name_hint,
+    const std::vector<ExprHandle>& dims,
+    Dtype dtype,
+    c10::optional<ExprHandle> initializer,
+    c10::optional<std::vector<ExprHandle>> strides,
+    c10::optional<ExprHandle> qscale,
+    c10::optional<ExprHandle> qzero) {
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/expr.cpp:439]: This was added in the previous PR, right? Wondering why clang-format didn't catch it there..

File: torch/csrc/jit/tensorexpr/kernel.h [hunk 2]
```diff
@@ -198,7 +198,6 @@ class TORCH_API TensorExprKernel {
   void genInputDebugNames();
   void runKernel(Stack& stack);
 
-  std::vector<DimArg> dimsFromSizes(const std::vector<ExprHandle>& sizes);
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/kernel.h:201]: This function wasn't defined anywhere!!! WTH :)

**Review by navahgar — APPROVED** [review]
LGTM

I didn't go through all the changes in tests, since they were mechanical.

File: torch/csrc/jit/tensorexpr/expr.cpp (hunk 1 above)
Comment (ZolotukhinM) [reply torch/csrc/jit/tensorexpr/expr.cpp:439]: Hm, yeah, maybe the previous PR breaks clang-format :)

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).

**Pull request closed without merging**